tests/unit/test_query.py

```diff
@@ -799,11 +990,114 @@ def test_pb_from_query_filter_key():
+                    PropertyFilter("category", "=", "Grocery"),
+                    PropertyFilter("price", ">", "100"),
+                ]
+            ),
+            Or(
+                [
+                    PropertyFilter("category", "=", "Stationery"),
+                    PropertyFilter("price", "<", "50"),
+                ]
+            ),
+            PropertyFilter("name", "=", "Jana"),
+        ]
+    )
+    query.OPERATORS = {"=": query_pb2.PropertyFilter.Operator.EQUAL}
+    pb = _pb_from_query(query)
+    filter = pb.filter.composite_filter
+
+    assert filter.op == query_pb2.CompositeFilter.Operator.AND
+    assert len(filter.filters) == 4
+
+    filter_1 = filter.filters[0].property_filter
+    assert filter_1.property.name == "name"
+    assert filter_1.value.string_value == "John"
+    assert filter_1.op == Query.OPERATORS.get("=")
+
+    filter_2 = filter.filters[1].composite_filter
+    assert len(filter_2.filters) == 2
+    assert filter_2.op == query_pb2.CompositeFilter.Operator.AND
+
+    filter_2_1 = filter_2.filters[0].property_filter
+    assert filter_2_1.property.name == "category"
+    assert filter_2_1.op == Query.OPERATORS.get("=")
+    assert filter_2_1.value.string_value == "Grocery"
+
+    filter_2_2 = filter_2.filters[1].property_filter
+    assert filter_2_2.property.name == "price"
+    assert filter_2_2.op == Query.OPERATORS.get(">")
+    assert filter_2_2.value.string_value == "100"
+
+    filter_3 = filter.filters[2].composite_filter
+    assert len(filter_3.filters) == 2
+    assert filter_3.op == query_pb2.CompositeFilter.Operator.OR
+
+    filter_3_1 = filter_3.filters[0].property_filter
+    assert filter_3_1.property.name == "category"
+    assert filter_3_1.op == Query.OPERATORS.get("=")
+    assert filter_3_1.value.string_value == "Stationery"
+
+    filter_3_2 = filter_3.filters[1].property_filter
+    assert filter_3_2.property.name == "price"
+    assert filter_3_2.op == Query.OPERATORS.get("<")
+    assert filter_3_2.value.string_value == "50"
+
+    filter_4 = filter.filters[3].property_filter
+    assert filter_4.property.name == "name"
+    assert filter_4.value.string_value == "Jana"
+    assert filter_4.op == Query.OPERATORS.get("=")
+
+
+def test_build_pb_for_and():
+
+    and_filter = And(
+        [
+            ("name", "=", "John"),
+            And(
+                [
+                    PropertyFilter("category", "=", "Grocery"),
+                    PropertyFilter("price", ">", "100"),
+                ]
+            ),
+            PropertyFilter("category", "=", "Grocery"),
+        ]
+    )
+    from google.cloud.datastore_v1.types import query as query_pb2
+
+    container_pb = (
+        query_pb2.Filter().composite_filter.filters._pb.add().composite_filter
+    )
+    pb = and_filter.build_pb(container_pb=container_pb)
+
+    assert pb.op == query_pb2.CompositeFilter.Operator.AND
+    assert len(pb.filters) == 3
+
+
+def test_and_query_composite_filter():
```

Comment:
Nit: Can we say `base_composite_filter` here?

Reply:
Updated.